internal/api/graphql/graph/resolver/component_instance_filter.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package resolver
+
+// This file will be automatically regenerated based on the schema, any resolver implementations
+// will be copied through when generating and any unknown code will be moved to the end.
+// Code generated by github.com/99designs/gqlgen version v0.17.51
+
+import (
+	"context"
+
+	"github.com/cloudoperators/heureka/internal/api/graphql/graph"
+	"github.com/cloudoperators/heureka/internal/api/graphql/graph/baseResolver"
+	"github.com/cloudoperators/heureka/internal/api/graphql/graph/model"
+)
+
+// ServiceName is the resolver for the serviceName field.
+func (r *componentInstanceFilterValueResolver) ServiceName(ctx context.Context, obj *model.ComponentInstanceFilterValue, filter *model.ServiceFilter) (*model.FilterItem, error) {
+	item, err := baseResolver.ServiceNameBaseResolver(r.App, ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+	item.FilterName = &baseResolver.ServiceFilterServiceName
+	return item, err
+}
+
+// SupportGroup is the resolver for the supportGroup field.
+func (r *componentInstanceFilterValueResolver) SupportGroupName(ctx context.Context, obj *model.ComponentInstanceFilterValue, filter *model.SupportGroupFilter) (*model.FilterItem, error) {
+	item, err := baseResolver.SupportGroupNameBaseResolver(r.App, ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+	item.FilterName = &baseResolver.ServiceFilterSupportGroupName
+	return item, err
+}
+
+// Ccrn is the resolver for the ccrn field.
+func (r *componentInstanceFilterValueResolver) Ccrn(ctx context.Context, obj *model.ComponentInstanceFilterValue, filter *model.ComponentInstanceFilter) (*model.FilterItem, error) {
+	item, err := baseResolver.CcrnBaseResolver(r.App, ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+	item.FilterName = &baseResolver.FilterDisplayCcrn
```

Comment:
Why is it called `FilterDisplayCcrn` and not `ServiceFilterCcrn` like the other names?

Reply:
@MR2011 I added these variables in internal/api/graphql/graph/baseResolver/common.go under the FilterDisplay "category" since they are no Services nor IssuesMatches since CCRN belongs to ComponentInstance. I can also rename them to ComponentInstanceCcrn if that is better.
These are the variables im referring to:
https://github.com/cloudoperators/heureka/blob/83641995fc9c22bad3b9602931363410f9fd0f42/internal/api/graphql/graph/baseResolver/common.go#L16-L42